ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CoreVerifier.kt

```diff
@@ -100,13 +103,26 @@ class CoreVerifier(val node: Node) {
                                 "than NotOnOrAfter element with value $notOnOrAfterValue.",
                         node = node)
         }
+
+        fun decryptAndValidateSchema(responseDom: Node) {
+            val encVerifier = EncryptionVerifier()
+
+            do {
+                Log.debugWithSupplier {
+                    "Starting a pass of decryption and schema validation" +
+                            " on the SAML Response."
+                }
+                SchemaValidator.validateSAMLMessage(responseDom)
+            } while (encVerifier.verifyAndDecryptResponse(responseDom))
```

Comment:
❓ Why is this in a loop? What is the exit condition for the loop?

Reply:
`verifyAndDecryptResponse` returns a Boolean on whether or not it found any encrypted elements. Because there may be encrypted elements within encrypted elements so we have to keep doing it until we find no more encrypted elements. That's super confusing though. Probably need to comment or javadoc that.